python/cudf/cudf/options.py

```diff
@@ -308,6 +308,21 @@ def _integer_and_none_validator(val):
     _make_contains_validator([False, True]),
 )
 
+_register_option(
+    "mode.pandas_debugging",
+    False,
+    textwrap.dedent(
+        """
+        If set to `False`, retains `cudf` specific behavior.
+        If set to `True`, enables pandas debugging mode,
+        which will raise a warning if the results from cudf
+        and pandas differ.
```

Comment:
Maybe something like this? We need to explain somehow that extra work is being done, in order to compare the results.
```suggestion
        If set to `True`, enables cudf.pandas debugging mode.
        When enabled, cudf code paths in cudf.pandas will
        also run with pandas and raise a warning if the
        results from cudf and pandas differ.
        If set to `False`, cudf.pandas debugging is disabled.
```